File: kasa_android/app/src/main/java/com/example/kasa/util/DateFormatter.kt
package com.example.kasa.util

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

object DateFormatter {
    private val locale = Locale.FRENCH

    fun formatDate(date: Date): String {
        return SimpleDateFormat("dd MMM yyyy", locale).format(date)
    }

    fun formatShort(date: Date): String {
        return SimpleDateFormat("dd MMM", locale).format(date)
    }

    fun formatTime(date: Date): String {
        return SimpleDateFormat("HH:mm", locale).format(date)
    }

    fun currentMonthName(): String {
        return SimpleDateFormat("MMMM yyyy", locale).format(Date())
    }

    fun currentMonthKey(): String {
        return SimpleDateFormat("yyyy-MM", Locale.US).format(Date())
    }

    fun formatRelative(date: Date): String {
        val diff = Date().time - date.time
        val minutes = diff / (1000 * 60)
        val hours = minutes / 60
        val days = hours / 24

        return when {
            minutes < 1 -> "À l'instant"
            minutes < 60 -> "Il y a ${minutes}m"
            hours < 24 -> "Il y a ${hours}h"
            days < 7 -> "Il y a ${days}j"
            else -> formatDate(date)
        }
    }
}

File: kasa/lib/utils/date_formatter.dart
import 'package:intl/intl.dart';

class DateFormatter {
  static String formatDate(DateTime date) {
    return DateFormat('dd MMM yyyy', 'fr_FR').format(date);
  }

  static String formatShort(DateTime date) {
    return DateFormat('dd MMM', 'fr_FR').format(date);
  }

  static String formatTime(DateTime date) {
    return DateFormat('HH:mm').format(date);
  }

  static String formatRelative(DateTime date) {
    final now = DateTime.now();
    final diff = now.difference(date);
    
    if (diff.inMinutes < 1) return "À l'instant";
    if (diff.inMinutes < 60) return 'Il y a ${diff.inMinutes} min';
    if (diff.inHours < 24) return 'Il y a ${diff.inHours}h';
    if (diff.inDays < 7) return 'Il y a ${diff.inDays}j';
    return formatDate(date);
  }

  static String currentMonth() {
    return DateFormat('MMMM yyyy', 'fr_FR').format(DateTime.now());
  }

  static String currentMonthKey() {
    final now = DateTime.now();
    return '${now.year}-${now.month.toString().padLeft(2, '0')}';
  }

  static String monthKey(DateTime date) {
    return '${date.year}-${date.month.toString().padLeft(2, '0')}';
  }

  static String monthName(String monthKey) {
    final parts = monthKey.split('-');
    final date = DateTime(int.parse(parts[0]), int.parse(parts[1]));
    return DateFormat('MMMM yyyy', 'fr_FR').format(date);
  }
}
